hibernate-core/src/main/java/org/hibernate/proxy/pojo/javassist/JavassistProxyFactory.java

```diff
@@ -93,7 +93,21 @@ public void postInstantiate(
 		javassist.util.proxy.ProxyFactory factory = new javassist.util.proxy.ProxyFactory() {
 			@Override
 			protected ClassLoader getClassLoader() {
-				return persistentClass.getClassLoader();
+				// This classloader does not take into account class space consistency
+				// i.e. if the persistenceClass can see the same interface types via
+				// a different ClassLoader hierarchy. In that case we could try some 
+				// risky magic with loadClass, but using findClass is much safer.
+				return new ClassLoader(persistentClass.getClassLoader()) {
```

Comment:
Ditto.  The main issue is the method currently relies solely on the classloader of the client bundle containing the persistent class.  That required the client bundle to import `org.hibernate.proxy` (for `HibernateProxy`) and `javassist.util.proxy` (for proxy interfaces -- I think...).  Providing visibility into the entire `ClassLoaderService` might be overkill, but I somewhat doubt it'd necessarily *hurt* anything.  Correct me if I'm wrong there -- uneducated guess.

Re: your other points, definitely agree -- we should try to find a "set once" approach, rather than instantiating this once per `Tuplizer`.

Thanks!